test/cmd_line/undo.test.ts

```diff
@@ -0,0 +1,25 @@
+import { getAndUpdateModeHandler } from '../../extension';
+import { commandLine } from '../../src/cmd_line/commandLine';
+import { ModeHandler } from '../../src/mode/modeHandler';
+import { assertEqualLines, cleanUpWorkspace, setupWorkspace } from '../testUtils';
+import { VimState } from '../../src/state/vimState';
+
+suite('Undo command', () => {
+  let modeHandler: ModeHandler;
+  let vimState: VimState;
+
+  setup(async () => {
+    await setupWorkspace();
+    modeHandler = await getAndUpdateModeHandler();
+    vimState = modeHandler.vimState;
+  });
+
+  teardown(cleanUpWorkspace);
+
+  test('undoes last action after insert mode', async () => {
+    await modeHandler.handleMultipleKeyEvents(['i', 'a', '<Esc>']);
+    await modeHandler.handleMultipleKeyEvents(['i', 'b', '<Esc>']);
+    await commandLine.Run('undo', vimState);
```

Comment:
Just refer to `modeHandler.vimState` here, rather than getting it in `setup`

Reply:
Will do 👍